apps/api/src/app/subscribers/usecases/update-subscriber-channel/update-subscriber-channel.usecase.ts

```diff
@@ -118,25 +118,55 @@ export class UpdateSubscriberChannel {
       return;
     }
 
+    let deviceTokens: string[] = [];
+
+    if (updatePayload.credentials?.deviceTokens) {
+      deviceTokens = this.unionDeviceTokens(
+        existingChannel.credentials.deviceTokens ?? [],
+        updatePayload.credentials.deviceTokens
+      );
+    }
+
     await this.invalidateCache.invalidateByKey({
       key: buildSubscriberKey({
         subscriberId: foundSubscriber.subscriberId,
         _environmentId: foundSubscriber._environmentId,
       }),
     });
 
-    const mergedChannel = Object.assign(existingChannel, updatePayload);
+    const mappedChannel: IChannelSettings = this.mapChannel(updatePayload, existingChannel, deviceTokens);
 
     await this.subscriberRepository.update(
       {
         _environmentId: environmentId,
         _id: foundSubscriber,
         'channels._integrationId': existingChannel._integrationId,
       },
-      { $set: { 'channels.$': mergedChannel } }
+      { $set: { 'channels.$': mappedChannel } }
     );
   }
 
+  private mapChannel(
+    updatePayload: Partial<IChannelSettings>,
+    existingChannel: IChannelSettings,
+    deviceTokens: string[]
+  ): IChannelSettings {
+    return {
+      _integrationId: updatePayload._integrationId || existingChannel._integrationId,
+      providerId: updatePayload.providerId || existingChannel.providerId,
+      credentials: { ...existingChannel.credentials, ...updatePayload.credentials, ...{ deviceTokens } },
+    };
+  }
+
+  private unionDeviceTokens(existingDeviceTokens: string[], updateDeviceTokens: string[]): string[] {
+    // in order to not have breaking change we will support [] update
+    if (updateDeviceTokens?.length === 0) return [];
```

Comment:
What if updateDeviceTokens is 0 but he does have existingDeviceTokens. Wouldn't this override the existing ones when used in the mapping?

Reply:
@ainouzgali 
do you mean it the array is empty it will erase the tokens?

Reply:
Yes, because if the update array is empty, you return an empty array into deviceTokens. But in the mapChannel function you use deviceToken after the spreading of the other credentials. So it will go through with deviceToken: [] even if existing deviceTokens has tokens

Reply:
yes you got it right, at the moment the API is supporting it and i am not sure if someone is using this 'hack' and in order to not break existing functionality i added this one.
once we will add a delete endpoint for the credentials we could remove this if. already added this note to task 2372.

Reply:
ohhhh its to update for delete !!! I missed the purpose 😂 I thought it was just for a case of empty update 
Sorry !